test/test_dispatch.py

```diff
@@ -0,0 +1,359 @@
+import torch._C as C
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+import itertools
+import unittest
+
+# TODO: Expand the dispatcher API to be a generic API for interfacing with
+# the dispatcher from Python!
+#
+# These are exhaustive tests for commutativity of dispatch behavior.  If you're
+# looking for more usage-info style tests, check op_registration_test.cpp
+#
+# Things not tested here:
+#   - Listeners
+#   - Top level namespace registrations
+#   - Fallback
+#   - Exotic overloads of CppFunction/schema
+#
+# Things not directly tested here:
+#   - Internal state of Dispatcher makes sense.  This is indirectly
+#     tested by the invariant testing
+
+class TestDispatch(TestCase):
+    namespace_index = 0
+
+    def test_all_invariants(self):
+        # Check that the regular stuff is OK!
+        C._dispatch_check_all_invariants()
+
+    # You probably don't want to call this directly; if your constructors
+    # don't commute, you can still run commute with a fixed ctor_order
+    # so that you can test that the destructors still commute
+    def run_ops(self, name, ops, ctor_order=None, dtor_order=None,
+                results=None, raises=False):
```

Comment:
nit: rename raises to allow_raises or expect_raises, depending on if passing in True and not raising is an error

Reply:
https://github.com/pytorch/pytorch/pull/35061/commits/cbc194d310c4fc7261986be5bd67cd97983bcfd7